### PR TITLE
Add an option ignore_first_spike, which explicit allows to ignore or consider the first spike during ISI computation

### PR DESCRIPTION
**Pull request by DrTaDa**

### CODE REVIEW

File: efel/api.py
Comment (wvangeit) [inline efel/api.py:102]: Is this a bool? (i.e. 0 == false, 1 is true)?. If yes, then let's limit this to these two values in the code below, and let's add it to the documentation

**Comment by DrTaDa** [comment]
@wvangeit My comments were there but stuck in "Pending" al of this time T_T

https://github.com/community/community/discussions/10369

File: efel/cppcore/LibV1.cpp
Comment (wvangeit) [reply efel/cppcore/LibV1.cpp:188]: I just wonder what 'retIgnore[0] < 0' means. Normally retignore will contain the python value (0 or 1?). 
Btw, are we sure retIgnore[0] will always exist, since the vector is uninitialized?

Can you also add a test for this setting.

**Comment by DrTaDa** [comment]
@wvangeit It was in case the user put a value < 0, but indeed it is not needed.

If the vector is uninitialized, retVal will be >=0 no ?

**Comment by wvangeit** [comment]
> @wvangeit It was in case the user put a value < 0, but indeed it is not needed.
> 
> If the vector is uninitialized, retVal will be >=0 no ?

You mean retVal < 0? Well it might be, but if it isn't for some reason we end up with a segmentation fault, it would be better to protect against it in this function.

**Comment by DrTaDa** [comment]
I added the test on retIgnore.size(), I am not sure if there is a better way

File: efel/cppcore/LibV1.cpp
Comment (wvangeit) [inline efel/cppcore/LibV1.cpp:188]: if something goes wrong with the ignore_first_spike feature, retVal will be -1 I think? I assume this will evaluate to True in C++. Is that what we really want? Shouldn't it fall back to the old behavior in that case, i.e. IgnoreFirstSpike = 1 ?

Comment (wvangeit) [reply efel/cppcore/LibV1.cpp:188]: Ah, but the size might be 0 in that case. It's still a bit implicit though. Maybe better retVal == 1?

Comment (DrTaDa) [reply efel/cppcore/LibV1.cpp]: Indeed

**Comment by AurelienJaquier** [comment]
Could this cause any unexpected behavior in other features that depend on ISI_values? I know that it would at least in `burst_mean_freq`. Although you can ignore this particular feature because I am going to change the code of the feature anyway in a future PR.

**Comment by DrTaDa** [comment]
@AurelienJaquier we kept the same default value, so it might but only if the user specifically request it.
I implemented this setting specifically for burst related computation, and it worked for me.

**Comment by DrTaDa** [comment]
@AurelienJaquier It doesn't bother me that other efeatures might have a different behaviour if a setting is changed. That is already the case for other settings. I would expect someone using this feature to know what he is doing.
(In our pipelines,  the settings are set per feature so it wouldn't cause us any issue).

**Comment by wvangeit** [comment]
@AurelienJaquier , do you mean that certain features assume that the first spike is ignored? If that's the case, I do indeed consider this a problem. Because in that case 'someone using this feature to know what he is doing' is not necessary true since they might not realize a certain features assumes this setting is not set.

**Comment by AurelienJaquier** [comment]
@wvangeit I just checked, and only burst_mean_freq would need to be modified after this merge, the other features would behave as expected.

**Comment by DrTaDa** [comment]
Let's wait for #250 to be merged and then I will fix burst_mean_freq here.

**Comment by AurelienJaquier** [comment]
For the new features I added in the last PR to the master branch, you just have to adapt the `burst_indices` feature.
It should be fairly easy, you just have to remove the comment [there](https://github.com/BlueBrain/eFEL/blob/master/efel/cppcore/LibV5.cpp#L3476), make it depend on `ignore_first_spike`, and change the initialisation of `first_ISI` and `count` into something like:
```
if (ignore_first_spike){
  int first_ISI=1, count = 1;
} else {
  int first_ISI=0, count = 0;
}
```

**Comment by AurelienJaquier** [comment]
I would like to rename ignore_first_spike into ignore_first_ISI for clarity, since it only affects the 1st ISI, and features depending on it, and not features depending on the 1st spike in general. Do you agree @DrTaDa ?

**Comment by AurelienJaquier** [comment]
Changes made in the last commit :
- renamed ignore_first_spike into ignore_first_ISI
- made burst features consistent with burst_indices when ignore_first_ISI is 0
- allow the new burst suite (strict_burst_* features) to also ignore the first ISI
- since the first ISI is ignored by default, I had to change the test results for the new burst features

**Review by AurelienJaquier — APPROVED** [review]
Is it ok to merge @wvangeit ?

**Review by wvangeit — APPROVED** [review]

**Pull request merged**
